packages/next/src/server/config.ts

```diff
@@ -412,12 +412,11 @@ function assignDefaults(
       )
     }
 
-    if (
-      images.path === imageConfigDefault.path &&
-      result.basePath &&
-      !pathHasPrefix(images.path, result.basePath)
-    ) {
-      images.path = `${result.basePath}${images.path}`
+    if (images.path === imageConfigDefault.path) {
+      const prefix = `${result.basePath || ''}${result.experimental?.nextUrlServerPrefix || ''}`
+      if (prefix && !pathHasPrefix(images.path, prefix)) {
+        images.path = `${prefix}${images.path}`
```

Comment:
Do we need to modify `images.path` and the loader? I was thinking we only need to do it once.